### PR TITLE
test/librados/test.cc: clean up EC pools' crush rules too

### PR DESCRIPTION
**Pull request by dmick**
SetUp was adding an erasure-coded pool, which automatically adds
a new crush rule named after the pool, but only removing the
pool.  Remove the crush rule as well.

Fixes: #13878
Signed-off-by: Dan Mick <dan.mick@redhat.com>

### CODE REVIEW

File: src/test/librados/test.cc [hunk 1]
```diff
@@ -241,14 +241,29 @@ int destroy_one_pool(const std::string &pool_name, rados_t *cluster)
...
-    }
-    rados_wait_for_latest_osdmap(*cluster);
+  if (ret) {
+    rados_shutdown(*cluster);
+    return ret;
+  }
+
+  char *cmd[2];
+
+  cmd[0] = (char *)(std::string("{\"prefix\": \"osd crush rule rm\", \"name\":\"") + pool_name + std::string("\"}")).c_str();
```
Comment (dachary) [inline src/test/librados/test.cc]: Could this be added in destroy_ec_profile instead ? The only other place this function is called won't be negatively impacted by the addtional crush rule rm. Or is there another reason for inlining it ?

**Comment by dachary** [comment]
@dmick could you please add a link to a successful run of smoke/basic/tasks/rados_api_tests.yaml ?

**Comment by liewegas** [comment]
I'll run it through this afternoon

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2015-11-26_14:02:28-smoke-wip-sage-testing2---basic-multi/

**Comment by liewegas** [comment]
it passed

File: src/test/librados/test.cc (hunk 1 above)
Comment (liewegas) [inline src/test/librados/test.cc]: This looks like a use-after-free.  Need to assign it to a local std::string

**Comment by dachary** [comment]
@dmick I can clean it up for you if you don't mind ?

**Comment by dachary** [comment]
continuing at https://github.com/ceph/ceph/pull/6788

**Comment by dmick** [comment]
Sorry, I saw none of this conversation somehow.

**Comment by dachary** [comment]
@dmick no worries ;-)

**Pull request closed without merging**